File: y2019/Day5/Day5.hs
import Text.Printf

import Intcode (interp)
import Utils (split)

main = do
    file <- readFile "day5/day5.txt"

    let code :: [Int] = map read (split file ',')

    interp code 0 -- provide 1 for p1, provide 5 for p2

File: y2019/go/day5.go
package main

import (
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func day5() {
	file, err := os.ReadFile("../input/day5.txt")
	if err != nil {
		log.Fatal(err)
	}

	var prog []int
	for _, i := range strings.Split(string(file), ",") {
		val, _ := strconv.Atoi(strings.TrimSpace(i))
		prog = append(prog, val)
	}

	fmt.Println("intcode started!")
	fmt.Println("type in 1 for the answer to p1, or 5 for the answer to p2")
	p := startIntcode(prog)
	p.run()
}
